### PR TITLE
zlib 1.2.11: support CMakeDeps generator + remove deprecated minizip option + modernize

### PR DESCRIPTION
**Pull request by SpaceIm**
Specify library name and version:  **lib/1.0**

This is also a good place to share with all of us **why you are submitting this PR** (specially if it is a new addition to ConanCenter): is it a dependency of other libraries you want to package? Are you the author of the library? Thanks!

---

- [ ] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [ ] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [ ] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [ ] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by SpaceIm** [comment]
@czoido Is there a way to change (in `package_info()`) prefix of CMake variables in `CMakeDeps`?

For example, `zlib_FOUND` is not the expected variable name: https://cmake.org/cmake/help/latest/module/FindZLIB.html

And I don't see variables like `ZLIB_INCLUDE_DIRS` and `ZLIB_LIBRARIES`.

File: recipes/zlib/1.2.11/conanfile.py [hunk 1]
```diff
@@ -1,22 +1,29 @@
...
 from conans import ConanFile, tools, CMake
-from conans.errors import ConanException
+import os
+
+required_conan_version = ">=1.43.0"
 
 
 class ZlibConan(ConanFile):
     name = "zlib"
-    version = "1.2.11"
```
Comment (Croydon) [inline recipes/zlib/1.2.11/conanfile.py:8]: The directory is called 1.2.11. I would either change the directory name as well or leave this here

Comment (SpaceIm) [reply recipes/zlib/1.2.11/conanfile.py:8]: I can rename the directoy to all, then merge 1.2.8 in this one.

Comment (SpaceIm) [reply recipes/zlib/1.2.11/conanfile.py:8]: done, the diff is still understandable

**Review by Croydon — APPROVED** [review]

**Comment by czoido** [comment]
> @czoido Is there a way to change (in `package_info()`) prefix of CMake variables in `CMakeDeps`? Something like `cmake_variable_prefix` property (might be different in Find and Config)?
> 

I think that we can partially fix the prefix in Conan by using the filename property for the prefix instead of the package name so that we generate `ZLIB_FOUND` and `ZLIB_VERSION` by default. Even have a look at adding `ZLIB_INCLUDE_DIRS` and `ZLIB_LIBRARIES`. But more than that will be difficult to address in the Conan side even adding new properties, because the variables defined by the official CMake modules are very heterogeneous and it would be impossible to satisfy them all... probably, if it's important to maintain full compatibility with those variables it should be addressed in the recipes with a build module or something similar...

**Comment by SpaceIm** [comment]
Sounds reasonable.

**Review by ericLemanissier — APPROVED** [review]

**Review by madebr — APPROVED** [review]

**Review by danimtb — APPROVED** [review]

**Review by uilianries — APPROVED** [review]
LGtm

**Review by czoido — APPROVED** [review]

**Pull request merged**